base/src/Data/Macaw/AbsDomain/JumpBounds.hs

```diff
@@ -386,8 +386,18 @@ exprRangePred info e = do
             -- Compare the range constraint with the output number of bits.
             -- If the bound on r covers at most the truncated number
             -- of bits, then we just pass it through.
-          , Just LeqProof <- testLeq (rangeWidth r) w ->
-            SomeRangePred r
+          -> case testLeq (rangeWidth r) w of
+              Just LeqProof -> SomeRangePred r
+              
+              -- Otherwise, our bound remains valid for the truncated number
+              -- of bits and we can tighten the bounds to at most @2^w-1@ (inclusive)
+              Nothing ->
+                let 
+                  truncUnsigned = fromInteger . unsignedClamp w . toInteger
+                  lowerBound = truncUnsigned $ rangeLowerBound r
```

Comment:
Actually, come to think of it, can the lower bound ever be lower than `0`? The type of `rangeLowerBound` is a `Natural`, after all, not an `Integer`.

As such, I think we could get away with not clamping the lower bounds here, which would make this part of the code slightly more efficient.

Reply:
I suppose one interesting edge case here would be if `lower_bound > 2^w - 1`. Then the clamping would tighten the bound to be `2^w -1`, right?

I wasn't too concerned with efficiency here, so if this is the only edge case maybe it should be handled differently?

Reply:
Ah, fair enough. In that case, I suppose we really do have to clamp the lower bound to be on the safe side.